### Importing the neccesary modules 


In [60]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [61]:
df = pd.read_csv('./Global+Superstore+Data.csv')

In [62]:
df.head()

Order Date      Segment  Market     Sales    Profit
0  31-07-2012     Consumer      US  2309.650  762.1845
1  05-02-2013    Corporate    APAC  3709.395 -288.7650
2  17-10-2013     Consumer    APAC  5175.171  919.9710
3  28-01-2013  Home Office      EU  2892.510  -96.5400
4  05-11-2013     Consumer  Africa  2832.960  311.5200

### Understanding the data

In [63]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Order Date  51290 non-null  object 
 1   Segment     51290 non-null  object 
 2   Market      51290 non-null  object 
 3   Sales       51290 non-null  float64
 4   Profit      51290 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.0+ MB


In [64]:
df.isnull().sum()
# Soo no null values data is already cleaned

Order Date    0
Segment       0
Market        0
Sales         0
Profit        0
dtype: int64

In [65]:
df.describe()

Sales        Profit
count  51290.000000  51290.000000
mean     246.490581     28.610982
std      487.565361    174.340972
min        0.444000  -6599.978000
25%       30.758625      0.000000
50%       85.053000      9.240000
75%      251.053200     36.810000
max    22638.480000   8399.976000

In [66]:
# Different market and customer segments
df['Market'].unique()

#  7 places = 'US' , 'APAC', 'EU', 'Africa', 'EMEA', 'LATAM', 'Canada'

array(['US', 'APAC', 'EU', 'Africa', 'EMEA', 'LATAM', 'Canada'],
      dtype=object)

In [67]:
df['Segment'].unique()

# 3  segements = ['Consumer', 'Corporate', 'Home Office']
# In total 21 unique segements

array(['Consumer', 'Corporate', 'Home Office'], dtype=object)

### Finding out the most profitable market segement

In [68]:
df['combinedMarketSegment'] = df['Market'] + '-' + df['Segment']

In [69]:
df.head()
# cross checking

Order Date      Segment  Market     Sales    Profit combinedMarketSegment
0  31-07-2012     Consumer      US  2309.650  762.1845           US-Consumer
1  05-02-2013    Corporate    APAC  3709.395 -288.7650        APAC-Corporate
2  17-10-2013     Consumer    APAC  5175.171  919.9710         APAC-Consumer
3  28-01-2013  Home Office      EU  2892.510  -96.5400        EU-Home Office
4  05-11-2013     Consumer  Africa  2832.960  311.5200       Africa-Consumer

In [70]:
# next converting the dataset
# converting the date into the required format
df['Order Date'] = pd.to_datetime(df['Order Date']).dt.to_period('m')
# to_period m convert to month

C:\Users\SAM\AppData\Local\Temp\ipykernel_19768\3464277095.py:3: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  df['Order Date'] = pd.to_datetime(df['Order Date']).dt.to_period('m')


In [71]:
df_agg = df.pivot_table(index='Order Date', values='Profit',columns='combinedMarketSegment',aggfunc='sum')
# as we are only concerned with the profit

In [72]:
df_agg.head()

combinedMarketSegment  APAC-Consumer  APAC-Corporate  APAC-Home Office  \
Order Date                                                               
2011-01                    2254.4898       -173.7264          230.2656   
2011-02                    4627.8531       3600.7776          617.5407   
2011-03                    2337.8319       3634.9365          205.1052   
2011-04                     633.6423         90.7521          603.8478   
2011-05                    3838.9500       1028.4489          102.0315   

combinedMarketSegment  Africa-Consumer  Africa-Corporate  Africa-Home Office  \
Order Date                                                                     
2011-01                        265.539           520.026             391.515   
2011-02                       -201.888           -31.689             825.606   
2011-03                        919.074           244.437              83.430   
2011-04                        381.315           941.751             417.618   
2011-05                        -39.981           490.650              39.180   

combinedMarketSegment  Canada-Consumer  Canada-Corporate  Canada-Home Office  \
Order Date                                                                     
2011-01                         447.33             12.57                 NaN   
2011-02                          26.43              9.24               87.99   
2011-03                          82.47               NaN               84.03   
2011-04                          55.08              6.51                 NaN   
2011-05                           0.87               NaN                 NaN   

combinedMarketSegment  EMEA-Consumer  ...  EMEA-Home Office  EU-Consumer  \
Order Date                            ...                                  
2011-01                    -2952.594  ...           392.526    2259.8670   
2011-02                      464.160  ...           368.070    -783.9945   
2011-03                      706.380  ...           -15.570    1273.7100   
2011-04                      121.353  ...          -456.510    2980.0065   
2011-05                      172.554  ...           -33.591     673.0080   

combinedMarketSegment  EU-Corporate  EU-Home Office  LATAM-Consumer  \
Order Date                                                            
2011-01                    841.3875        2534.568       540.44312   
2011-02                   2234.4060         162.255      1227.86616   
2011-03                   2206.5960         269.745      2427.39612   
2011-04                    981.1470         130.371      1368.82948   
2011-05                   1719.0600         443.310      1749.39200   

combinedMarketSegment  LATAM-Corporate  LATAM-Home Office  US-Consumer  \
Order Date                                                               
2011-01                     1021.05924          -26.59744    2465.8368   
2011-02                      814.03020         -356.80440    1853.5541   
2011-03                      178.66400          769.88400     204.2007   
2011-04                      872.64068         -554.22952    1434.8636   
2011-05                      134.87252         -397.95600     833.9433   

combinedMarketSegment  US-Corporate  US-Home Office  
Order Date                                           
2011-01                    643.3689       1233.9329  
2011-02                    421.9256        553.9467  
2011-03                    127.4696       -268.9314  
2011-04                    494.7315       2723.4358  
2011-05                   2912.7976        145.7781  

[5 rows x 21 columns]

In [73]:
df_agg.shape

(48, 21)

In [74]:
# Train - test Split
train_len = 42
train_df = df_agg[0 : train_len]
test_df = df_agg[train_len :]

In [75]:
train_df.shape

(42, 21)

In [76]:
test_df.shape
# combined is 48

(6, 21)

In [78]:
# Finding our the covariance
mean = np.mean(train_df)
std = np.std(train_df)
convariance_df =pd.DataFrame(mean)
convariance_df['std'] = std
convariance_df['covariance'] = std / mean
convariance_df.sort_values(by='covariance',ascending=True)
convariance_df

c:\sahil\code\upgrad\assignment\Global_Mart\env\Lib\site-packages\numpy\core\fromnumeric.py:3462: FutureWarning: In a future version, DataFrame.mean(axis=None) will return a scalar mean over the entire DataFrame. To retain the old behavior, use 'frame.mean(axis=0)' or just 'frame.mean()'
  return mean(axis=axis, dtype=dtype, out=out, **kwargs)


0          std  covariance
combinedMarketSegment                                      
APAC-Consumer          4400.894243  2300.457687    0.522725
APAC-Corporate         2574.919807  1364.837734    0.530051
APAC-Home Office       1511.088314  1523.508658    1.008219
Africa-Consumer         957.707000  1254.932072    1.310351
Africa-Corporate        412.617571   780.566850    1.891744
Africa-Home Office      377.221071   759.322203    2.012937
Canada-Consumer         225.987632   282.555788    1.250315
Canada-Corporate         90.980294   162.493114    1.786025
Canada-Home Office      118.003750   279.632866    2.369695
EMEA-Consumer           423.960286  1124.552711    2.652495
EMEA-Corporate          182.642643  1160.698430    6.355024
EMEA-Home Office         84.231366   651.283095    7.732073
EU-Consumer            3699.977143  2202.282289    0.595215
EU-Corporate           2216.299429  1600.336696    0.722076
EU-Home Office         1224.456536  1148.627937    0.938072
LATAM-Consumer         2295.555697  1569.632686    0.683770
LATAM-Corporate        1122.633016   990.360880    0.882177
LATAM-Home Office       818.398941   957.275713    1.169693
US-Consumer            2686.740912  2715.031412    1.010530
US-Corporate           1754.199083  1880.200775    1.071829
US-Home Office         1132.065762  1272.476439    1.124030

### Inference
#### APAC - Consumer has the lowest covariance 0.522725
hence we should focus on this sector as we want the least variance in our profits